File: ANDROID DEV/Crew/app/src/main/java/com/horseman/crew/MainActivity.java
package com.horseman.crew;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;


import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    ListView listView;

    public static List<PersonModel> personModelList = new ArrayList<>();
    PersonModel personModel;
    CustomAdapter customAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportActionBar().setTitle("Crew Members");
        //initialize variable
        listView = findViewById(R.id.listView);

        //now fetching data

        fetchdata();

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                startActivity(new Intent(getApplicationContext(), Details.class).putExtra("position", position));
            }
        });


    }

    private void fetchdata() {
        String url = "https://api.spacexdata.com/v4/crew";

        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        try {
                            JSONArray jsonArray = new JSONArray(response);
                            for (int i = 0; i < jsonArray.length(); i++) {
                                JSONObject jsonObject = jsonArray.getJSONObject(i);

                                String name = jsonObject.getString("name");
                                String agency = jsonObject.getString("agency");
                                String status = jsonObject.getString("status");
                                String image = jsonObject.getString("image");
                                String wikipedia = jsonObject.getString("wikipedia");
                                System.out.println(wikipedia);

                                personModel = new PersonModel(image, name, agency, status, wikipedia);
                                personModelList.add(personModel);
                            }
                            customAdapter = new CustomAdapter(MainActivity.this, personModelList);
                            listView.setAdapter(customAdapter);
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });
        RequestQueue requestQueue = Volley.newRequestQueue(this);
        requestQueue.add(stringRequest);
    }
}

File: ANDROID DEV/Memes/app/src/main/java/com/horseman/memes/MainActivity.kt
package com.horseman.memes

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.bumptech.glide.Glide

class MainActivity : AppCompatActivity() {
     lateinit var imageView: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.memeImageView)
        fetchData()
    }

    private fun fetchData() {
        val queue = Volley.newRequestQueue(this);
        val url = "https://meme-api.herokuapp.com/gimme"

        val jsonObjectRequest = JsonObjectRequest(Request.Method.GET, url, null,
            { response ->
                val url = response.getString("url")
                //val image = findViewById<ImageView>(R.id.memeImageView)
                Glide.with(this).load(url).into(imageView)
            },
            {
                Toast.makeText(this, "failed", Toast.LENGTH_LONG).show()
            })
        queue.add(jsonObjectRequest)
    }

    fun shareMeme(view: View) {}
    fun nextMeme(view: View) {}
}